pkg/executor/infoschema_reader.go

```diff
@@ -563,151 +587,275 @@ func (e *memtableRetriever) updateStatsCacheIfNeed() bool {
 	return false
 }
 
-func (e *memtableRetriever) setDataFromTables(ctx context.Context, sctx sessionctx.Context, schemas []model.CIStr) error {
-	useStatsCache := e.updateStatsCacheIfNeed()
-	checker := privilege.GetPrivilegeManager(sctx)
-
-	var rows [][]types.Datum
-	createTimeTp := mysql.TypeDatetime
-	loc := sctx.GetSessionVars().TimeZone
-	if loc == nil {
-		loc = time.Local
+func getMatchSchemas(
+	extractor base.MemTablePredicateExtractor,
+	is infoschema.InfoSchema,
+) []model.CIStr {
+	ex := extractor.(plannercore.TableSchemaSelector)
```

Comment:
```suggestion
	ex , ok := extractor.(plannercore.TableSchemaSelector)
```

will panic if not `ok`. If you have confident to make sure `TableSchemaSelector` is always implemented it's no need to check `if ex != nil` in line 595?